Gems/PhysX/Code/Tests/CharacterControllerTests.cpp

```diff
@@ -159,6 +159,115 @@ namespace PhysX
+            characterGameplayConfiguration->m_groundDetectionBoxHeight = 0.05f;
+
+            transform = gameplayEntity->CreateComponent<AzFramework::TransformComponent>();
+            characterComponent = gameplayEntity->CreateComponent<CharacterControllerComponent>(
+                AZStd::move(characterConfiguration), AZStd::move(characterShapeConfiguration));
+            gameplayComponent = gameplayEntity->CreateComponent<CharacterGameplayComponent>(AZStd::move(characterGameplayConfiguration));
+
+            transform->SetWorldTM(AZ::Transform::Identity());
+        }
+        gameplayEntity->Init();
+        gameplayEntity->Activate();
+
+        // Let scene run for a few moments so the entity can be manipulated by gravity from the gameplay component
+        auto startTransform = transform->GetWorldTM();
+
+        int duration = 10;
+        float totalTime = 0.0f;
+        float timeStep = AzPhysics::SystemConfiguration::DefaultFixedTimestep;
+
+        if (auto* physXSystem = GetPhysXSystem())
+        {
+            for (int i = 0; i < duration; i++)
+            {
+                physXSystem->Simulate(timeStep);
+                totalTime += timeStep;
+            }
+        }
+
+        AZ::Transform endTransform = transform->GetWorldTM();
+
+        EXPECT_FALSE(endTransform.IsClose(startTransform));
+        EXPECT_NEAR(expected_gravity, gameplayComponent->GetGravityMultiplier(), AZ::Constants::Tolerance);
+    }
+
+    TEST_F(PhysXDefaultWorldTest, CharacterController_GamePlayComponenet_GravitySetsWhileMoving)
+    {
+        PhysX::CharacterControllerComponent* characterComponent = nullptr;
+        PhysX::CharacterGameplayComponent* gameplayComponent = nullptr;
+        AzFramework::TransformComponent* transform = nullptr;
+
+        float expected_gravity = 2.5f;
+
+        // Create character
+        auto gameplayEntity = AZStd::make_unique<AZ::Entity>("GameplayEntity");
+        {
+            auto characterConfiguration = AZStd::make_unique<Physics::CharacterConfiguration>();
+            auto characterShapeConfiguration = AZStd::make_unique<Physics::CapsuleShapeConfiguration>();
+            auto characterGameplayConfiguration = AZStd::make_unique<PhysX::CharacterGameplayConfiguration>();
+
+            characterShapeConfiguration->m_height = 1.5f;
+            characterShapeConfiguration->m_radius = 0.5f;
+
+            characterGameplayConfiguration->m_gravityMultiplier = 1.5f;
+            characterGameplayConfiguration->m_groundDetectionBoxHeight = 0.05f;
+
+            transform = gameplayEntity->CreateComponent<AzFramework::TransformComponent>();
+            characterComponent = gameplayEntity->CreateComponent<CharacterControllerComponent>(
+                AZStd::move(characterConfiguration), AZStd::move(characterShapeConfiguration));
+            gameplayComponent = gameplayEntity->CreateComponent<CharacterGameplayComponent>(AZStd::move(characterGameplayConfiguration));
+
```

Comment:
update: This code all looks nearly identical in both tests, could we extract it to a utility function (or move to the fixture setup) to avoid so much duplication? Thanks!

Reply:
Will tackle this once I get clarity on above questions.

Reply:
+1

Reply:
Going to refactor this into either a helper or a test basis.

Reply:
Nice one thanks! 👍

Reply:
Backlogged into internal tracking and simplified scope of this test into a single test.